hera_mc/cm_hookup.py

```diff
@@ -106,21 +105,48 @@ def __get_part_types_found(self, huco, part_types_found):
             part_types_found.append(part_type)
         return part_types_found
 
-    def __get_pols_to_do(self, part, port, check_pol=False):
-        all_pols = ['e', 'n']
-        if port[0].lower() in all_pols:
-            pols = [port[0].lower()]
-        elif port.lower() in ['a', 'b']:
-            pols = [part[-1].lower()]
-        else:
-            pols = all_pols
-        if check_pol is not False:
-            if check_pol in pols:
-                pols = True
+    def __get_pols_to_do(self, part, port_query):
+        """
+        Given the current part and port_query (which is either 'all', 'e', or 'n')
+        this figures out which pols to do.  Basically, given 'all' and part it
+        figures out whether to return ['e'], ['n'], ['e', 'n']
+
+        Parameter:
+        -----------
+        part:  current part dossier
+        port_query:  the ports that were requested.
+        """
+        single_pol_parts_paper = ['RI', 'RO', 'CR']
+        pq = port_query.lower()
+        if pq == 'all':  # Need to figure out if return 'e', 'n' or both
+            if part['part'].hpn[:2].upper() in single_pol_parts_paper:
+                pols = [part['part'].hpn[-1].lower()]
             else:
-                pols = False
+                pols = PC.both_pols
+        elif pq in PC.both_pols:
+            pols = [ports]
+        else:
+            pols = PC.both_pols
```

Comment:
I'm not sure if this is the best default behavior. The `port_query` passed to this function is either a hardcoded `all` value, or the value of the `--port` argument to `parts.py`. So if I run `parts.py --port=blooga` the program will basically ignore my `--port` argument altogether. I think it's better to be a bit stricter and raise an error if the user provides a value that we don't recognize.